### PR TITLE
New jsonstat2 serializer with more metadata.

### PR DESCRIPTION
**Pull request by oyessb**
Created models from PxApi2 open api spec with openapi-generator-cli.
Notes are moved out of extension to standard json-stat2 note.
New extenions on root, dimension and category are used to describe if a note is mandatory.
Added new tests for jsonstat2.

### CODE REVIEW

**Review by runejo — APPROVED** [review]
Made som adjustments to the tests, thats all 👍

File: PCAxis.Serializers/JsonStat2Serializer.cs [hunk 1]
```diff
@@ -48,269 +33,366 @@ public class JsonStat2Serializer : IPXModelStreamSerializer
...
-            string result = JsonConvert.SerializeObject(jsonStat,new DecimalJsonConverter());
- 
-            return result;
+                noteIndex++;
+            }
         }
-        
 
-        public void Serialize(PXModel model, Stream stream)
+        private void AddContacts(Dataset dataset, string contactString)
```
Comment (likp) [inline PCAxis.Serializers/JsonStat2Serializer.cs:273]: We do a slightly diffrent mapping of the contact information in the custom json format https://github.com/statisticssweden/PxWeb/blob/860c2e35124d684e6ec833d8778a0ee83403325e/PxWeb/Mappers/TableMetadataResponseMapper.cs#L254 where we utilyze the a phared contect object if it exists. (Only exists when CNMM and PX files that that have the the same format as the SQL-Parser.)

Comment (oyessb) [reply PCAxis.Serializers/JsonStat2Serializer.cs]: Changed how contact is set. It now uses Paxiom.Contact if it exist. 
Multiple contacts defined in ContInfo.Contact will be splitted into seperate contacts in root extenion/contact.

**Review by likp — APPROVED** [review]

**Review by runejo — APPROVED** [review]

**Comment by likp** [comment]
@oyessb your latest change are fine, but there appears to be some conflicts that has to be resolved.

**Pull request merged**